Framework/Algorithms/src/PolarizationCorrections/PolarizerEfficiency.cpp

```diff
@@ -0,0 +1,204 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "MantidAlgorithms/PolarizationCorrections/PolarizerEfficiency.h"
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspaceUnitValidator.h"
+#include "MantidAlgorithms/PolarizationCorrections/PolarizationCorrectionsHelpers.h"
+#include "MantidAlgorithms/PolarizationCorrections/SpinStateValidator.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/ListValidator.h"
+
+#include <boost/algorithm/string/join.hpp>
+#include <filesystem>
+
+namespace Mantid::Algorithms {
+// Register the algorithm into the algorithm factory
+DECLARE_ALGORITHM(PolarizerEfficiency)
+
+using namespace Kernel;
+using namespace API;
+
+namespace PropertyNames {
+static const std::string INPUT_WORKSPACE = "InputWorkspace";
+static const std::string ANALYSER_EFFICIENCY = "AnalyserEfficiency";
+static const std::string SPIN_STATES = "SpinStates";
+static const std::string OUTPUT_WORKSPACE = "OutputWorkspace";
+static const std::string OUTPUT_FILE_PATH = "OutputFilePath";
+} // namespace PropertyNames
+
+namespace {
+static const std::string FILE_EXTENSION = ".nxs";
+
+void validateInputWorkspace(MatrixWorkspace_sptr const &ws, std::string const &propertyName,
+                            std::map<std::string, std::string> &errorList) {
+  if (ws == nullptr) {
+    errorList[propertyName] = "All input workspaces must be of type MatrixWorkspace.";
+    return;
+  }
+  if (ws->getNumberHistograms() != 1) {
```

Comment:
There is a benefit to adding as many as possible to the error list before returning,  as the red asterisks in the Algorithm Dialog are set based on this. So it makes it more clear there that there are multiple issues.

Reply:
I think maybe my comment is a bit confusing, sorry. At this point in the code we're in helper method `validateInputWorkspace` to check for errors with a single property, rather than in the `validateInputs` method. If we return early from `validateInputWorkspace` then I think we would still check the rest of the properties. I'm just thinking that when we're validating a single property (so we're only going to get one asterisk out of it, regardless of the number of problems found) we can technically stop checking as soon as the first problem is encountered.